Connection tracking split arch changes

hirentha opened the pull request:
Ct Test cases

    Signed-off-by: Hiren Thakkar <hiren.thakkar@intel.com>

hirentha commented:
DPDK CT flow dump is not included. once this will merge to main will do that seprately.

sjwahub commented on tests/ct_connection_tap_link_ports_with_scapy_icmp.py:
```diff
@@ -0,0 +1,143 @@
+        try:
+            verify_packets(self, pkt, device_number=1, ports=[port_ids[self.config_data['traffic']['receive_port'][1]][1]]) 
+        
+            log.passed(f"PASS: Verification of icmp packets B->A passed, icmp packet received")
```
In log.passed(), the PASS is not necessary. Please remove it.

sjwahub commented on tests/ct_connection_tap_link_ports_with_scapy_icmp.py:
```diff
@@ -0,0 +1,143 @@
+            log.passed(f"PASS: Verification of icmp packets B->A passed, icmp packet received")
+        except Exception as err:
+            self.result.addFailure(self, sys.exc_info())
+            log.failed(f"FAIL: Verification of icmp packet B->A sent failed with exception {err}")
```
The "FAIL" is not necessary. Please remove it.

hirentha commented:
logs for 11tcs run.
[ct_all_test_logs.txt](https://github.com/ipdk-io/ptf-tests/files/10721948/ct_all_test_logs.txt)

sjwahub commented on tests/ct_connection_tap_link_ports_with_scapy_udp.py:
```diff
@@ -0,0 +1,143 @@
+        try:
+            verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][1]][1]]) 
+        
+            log.passed(f"PASS: Verification of udp packets B->A passed, udp packet received")
```
remove the word PASS in log.passed()

sjwahub commented on tests/ct_connection_tap_link_ports_with_scapy_udp.py:
```diff
@@ -0,0 +1,143 @@
+            log.passed(f"PASS: Verification of udp packets B->A passed, udp packet received")
+        except Exception as err:
+            self.result.addFailure(self, sys.exc_info())
+            log.failed(f"FAIL: Verification of udp packet B->A sent failed with exception {err}")
```
please remove FAIL in log.failed()

sjwahub commented on tests/ct_connection_vhost_ports_with_netperf_udp.py:
```diff
@@ -0,0 +1,176 @@
+from scapy.packet import *
+from scapy.fields import *
+from scapy.all import *
+
```
The line 34-37 scapy are never used. Only netperf traffic. Suggest to remove them for clean code.

sjwahub commented on tests/ct_connection_vhost_ports_with_ping_icmp.py:
```diff
@@ -0,0 +1,165 @@
+# scapy related imports
+from scapy.packet import *
+from scapy.fields import *
+from scapy.all import *
```
The line 34 -37 are never used. Suggest to remove them for clean code.

sjwahub commented on tests/ct_connection_vhost_ports_with_netperf_udp.py:
```diff
@@ -0,0 +1,176 @@
+        test_params = test_params_get()
+        config_json = test_params['config_json']
+        self.dataplane = ptf.dataplane_instance
+        ptf.dataplane_instance = ptf.dataplane.DataPlane(config)
```
As no scapy involved, suggest to remove line 57-58

sjwahub commented on tests/ct_connection_vhost_ports_with_netperf_udp.py:
```diff
@@ -0,0 +1,176 @@
+        
+        for port_id, ifname in config["port_map"].items():
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
```
As no scapy involved, it's redundant to add port into dataplane. Suggest to remove line 77-79

sjwahub commented on tests/ct_connection_vhost_ports_with_netperf_udp.py:
```diff
@@ -0,0 +1,176 @@
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(f"FAIL: failed to start netperf")
+
+        self.dataplane.kill()
```
With previous comment, as no scapy involved, dataplane related code is removed. This line is needed to remove too.

sjwahub commented on tests/ct_connection_vhost_ports_with_ping_icmp.py:
```diff
@@ -0,0 +1,165 @@
+        test_params = test_params_get()
+        config_json = test_params['config_json']
+        self.dataplane = ptf.dataplane_instance
+        ptf.dataplane_instance = ptf.dataplane.DataPlane(config)
```
In this case, only ping is used, no dataplane traffic involved. Suggest to remove line 57-58.

sjwahub commented on tests/ct_connection_vhost_ports_with_ping_icmp.py:
```diff
@@ -0,0 +1,165 @@
+        
+        for port_id, ifname in config["port_map"].items():
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
```
In this case, only ping is used, no dataplane traffic involved. Suggest to remove line 77-79.

sjwahub commented on tests/ct_connection_vhost_ports_with_ping_icmp.py:
```diff
@@ -0,0 +1,165 @@
+
+
+
+        self.dataplane.kill()
```
Same reason. Suggest to remove this line

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
             verify_no_packet(self, pkt, port_ids[self.config_data['traffic']['receive_port'][2]][1])
 
-            print(f"PASS: Verification of data packet check passed : No traffic between A->B")
+            log.passed(f"PASS: Verification of data packet check passed : No traffic between A->B")
```
The PASS need to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
+            log.failed(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
```
The "FAIL" is needed to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
             verify_no_packet(self, pkt, port_ids[self.config_data['traffic']['receive_port'][3]][1])
 
-            print(f"PASS: Verification of date packet check passed : No traffic between B->A")
+            log.passed(f"PASS: Verification of date packet check passed : No traffic between B->A")
```
The PASS is needed removed from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of data packet sent failed with exception {err}: B->A")
+            log.failed(f"FAIL: Verification of data packet sent failed with exception {err}: B->A")
```
The "FAIL" needed to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])              
-            print(f"PASS: Verification of packets passed, Syn packet received")
+            log.passed(f"PASS: Verification of packets passed, Syn packet received")
```
The PASS needed to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of Syn packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of Syn packet sent failed with exception {err}")
```
The FAIL needed to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][2]][1]])
-            print(f"PASS: Verification of packets passed, Syn packet received")
+            log.passed(f"PASS: Verification of packets passed, Syn packet received")
```
The PASS needed to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of Syn packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of Syn packet sent failed with exception {err}")
```
The FAIL needed to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][1]][1]]) 
         
-            print(f"PASS: Verification of packets passed, SynAck packet received")
+            log.passed(f"PASS: Verification of packets passed, SynAck packet received")
```
The PASS needed to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
```
The FAIL needed to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
 
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])  
-            print(f"PASS: Verification of packets passed, ACK packet received")
+            log.passed(f"PASS: Verification of packets passed, ACK packet received")
```
The PASS needed to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of ACK packets sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of ACK packets sent failed with exception {err}")
```
The FAIL need to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][3]][1]])
 
-            print(f"PASS: Verification of packets passed, SynAck packet received")
+            log.passed(f"PASS: Verification of packets passed, SynAck packet received")
```
The PASS need to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
```
The FAIL need to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][2]][1]])
 
-            print(f"PASS: Verification of packets passed, ACK packet received")
+            log.passed(f"PASS: Verification of packets passed, ACK packet received")
```
The PASS need to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of ACK packets sent failed with exception {err}") 
+            log.failed(f"FAIL: Verification of ACK packets sent failed with exception {err}") 
 
```
The FAIL need to remove from log.failed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])  
-            print(f"PASS: Verification of data packet passed : A->B")
+            log.passed(f"PASS: Verification of data packet passed : A->B")
```
The PASS need to remove from log.passed()

sjwahub commented on tests/ct_multiple_connection_tap_link_ports.py:
```diff
@@ -80,318 +81,342 @@ def runTest(self):
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
+            log.failed(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
```
The FAIL need to remove from log.failed()

sjwahub commented:
1. The word PASS and FAIL need to remove from log.passed() and log.failed() respectfully. 
2. New test cases needed to add into test_to_run.txt

dkaranjai commented on pre_test.sh:
```diff
@@ -53,6 +53,10 @@ $IPDK_RECIPE/install/sbin/ovs-vswitchd --detach --no-chdir unix:$RUN_OVS/var/run
 $IPDK_RECIPE/scripts/dpdk/set_hugepages.sh
 
+rm -rf /usr/share/stratum/dpdk/dpdk_skip_p4.conf
+cp common/p4c_artifacts/pna_tcp_connection_tracking/dpdk_skip_p4.conf /usr/share/stratum/dpdk/ 
```
/usr/share/stratum/dpdk - will this path always exist? or do you want to check if it exists, if not then create it?

dkaranjai commented on tests/ct_connection_tap_link_ports_with_scapy_icmp.py:
```diff
@@ -0,0 +1,143 @@
+
+
+    def runTest(self):
+        if not test_utils.gen_dep_files_p4c_dpdk_pna_tdi_pipeline_builder(self.config_data):
```
step comment missing

dkaranjai commented on tests/ct_connection_tap_link_ports_with_scapy_icmp.py:
```diff
@@ -0,0 +1,143 @@
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to generate P4C artifacts or pb.bin")
+        
+        if not gnmi_ctl_set_and_verify(self.gnmictl_params):
```
step comment missing

dkaranjai commented on tests/ct_connection_tap_link_ports_with_scapy_icmp.py:
```diff
@@ -0,0 +1,143 @@
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
+
+        if not p4rt_ctl.p4rt_ctl_set_pipe(self.config_data['switch'], self.config_data['pb_bin'], self.config_data['p4_info']):
```
step comment missing

dkaranjai commented on tests/ct_connection_tap_link_ports_with_scapy_icmp.py:
```diff
@@ -0,0 +1,143 @@
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to set pipe")
+
+        table = self.config_data['table'][0]
```
step comment missing

dkaranjai commented on tests/ct_connection_tap_link_ports_with_scapy_icmp.py:
```diff
@@ -0,0 +1,143 @@
+        send_packet(self, port_ids[self.config_data['traffic']['send_port'][0]], pkt)
+        try:
+            verify_packets(self, pkt, device_number=1, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])              
+            log.passed(f"PASS: Verification of icmp packets A->B passed, icmp packet received")
```
remove PASS:

dkaranjai reviewed (changes requested):
General comments:
1. Step comment missing in multiple cases
2. log.passed and log.failed dont need "PASS" and "FAIL" in the message
3. use "black" tool once to reformat the code

stadju reviewed (commented):
In connection test, after FIN and RST, maybe you can send data packet to make sure the connection is closed.
2/13/2023,23:18:59  INFO: Sending FIN packet: A->B
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, FIN packet received
2/13/2023,23:18:59  INFO: Sending FIN+ACK packet: B->A
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, FIN+ACK packet received
2/13/2023,23:18:59  INFO: Sending FIN packet: B->A
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, FIN packet received B->A
2/13/2023,23:18:59  INFO: Sending ACK packet: A->B
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, ACK packet received
===> send data packet
2/13/2023,23:18:59  INFO: Connection Reset
2/13/2023,23:18:59  INFO: Sending RST packet: A->B
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, RST packet received
===> send data packet

hirentha commented:
Addressed all comments and amended all changes. Please review it. 

====================
Summary
====================
ct_single_connection_tap_link_ports : PASSED
ct_multiple_connection_tap_link_ports : PASSED
ct_rule_del_add_test : PASSED
nt_ct_unsupported_arguments_values : PASSED
ct_single_connection_vhost_ports_with_netperf : PASSED
ct_multiple_connection_vhost_ports_with_netperf : PASSED
ct_timer_test : PASSED
ct_connection_tap_link_ports_with_scapy_icmp : PASSED
ct_connection_tap_link_ports_with_scapy_udp : PASSED
ct_connection_vhost_ports_with_ping_icmp : PASSED
ct_connection_vhost_ports_with_netperf_udp : PASSED

[ct_all_test_logs.txt](https://github.com/ipdk-io/ptf-tests/files/10730676/ct_all_test_logs.txt)

dkaranjai commented:
Applied formatting of code using black

dkaranjai reviewed (approved):
LGTM
